api/src/repositories/cartRepo.ts

```diff
@@ -0,0 +1,253 @@
+/**
+ * Repository for cart data access
+ */
+
+import { getDatabase, DatabaseConnection } from '../db/sqlite';
+import { Cart } from '../models/cart';
+import { CartItem } from '../models/cartItem';
+import { Product } from '../models/product';
+import { handleDatabaseError, NotFoundError, ValidationError } from '../utils/errors';
+import { buildInsertSQL, objectToCamelCase, DatabaseRow } from '../utils/sql';
+
+export interface CartItemWithProduct extends CartItem {
+  product: Product;
+}
+
+export class CartRepository {
+  private db: DatabaseConnection;
+
+  constructor(db: DatabaseConnection) {
+    this.db = db;
+  }
+
+  /**
+   * Get cart by session ID
+   */
+  async getCartBySessionId(sessionId: string): Promise<Cart | null> {
+    try {
+      const row = await this.db.get<DatabaseRow>(
+        'SELECT * FROM carts WHERE session_id = ?',
+        [sessionId],
+      );
+      return row ? objectToCamelCase<Cart>(row) : null;
+    } catch (error) {
+      handleDatabaseError(error);
+    }
+  }
+
+  /**
+   * Create a new cart for a session
+   */
+  async createCart(sessionId: string): Promise<Cart> {
+    try {
+      const now = new Date().toISOString();
+      const { sql, values } = buildInsertSQL('carts', {
+        sessionId,
+        createdAt: now,
+        updatedAt: now,
+      });
+      
+      const result = await this.db.run(sql, values);
```

Comment:
Unused variable result.
```suggestion
      const result = await this.db.run(sql, values);
      if (!result || (typeof result.changes === 'number' && result.changes < 1)) {
        throw new ValidationError('Failed to insert cart');
      }
```